### PR TITLE
IR for chisq, fet, ctt, hwe with simpler chisq implementation

### PR DESCRIPTION
**Pull request by jbloom22**
- ir for chisq, fet, ctt, hwe
- using nan instead of missing
- direct implementation of chisq

This is first half of closed #3825, leaving all python docs and interface as is.

To cross-check, I used R and https://www.cog-genomics.org/software/stats

### CODE REVIEW

File: src/test/scala/is/hail/expr/ir/MathFunctionsSuite.scala [hunk 1]
```diff
@@ -94,4 +97,59 @@ class MathFunctionsSuite extends TestNGSuite {
       Uniroot("x", Ref("x", tfloat), NA(tfloat), F64(0)),
       null)
   }
+
+  @Test def chisq() {
```
Comment (catoverdrive) [inline src/test/scala/is/hail/expr/ir/MathFunctionsSuite.scala]: The other tests have been using the `@DataProvider` annotation to do this kind of parameterized testing. Do you want to do that here as well?

Comment (jbloom22) [reply src/test/scala/is/hail/expr/ir/MathFunctionsSuite.scala]: Is this what you had in mind? Do you prefer it?
```
  @DataProvider(name = "chisq")
  def chisqData(): Array[Array[Any]] = Array(
    Array(0, 0, 0, 0, Double.NaN, Double.NaN),
    Array(0, 1, 1, 1, 0.38647623077123266, 0.0),
    Array(1, 0, 1, 1, 0.38647623077123266, Double.PositiveInfinity),
    Array(1, 1, 0, 1, 0.38647623077123266, Double.PositiveInfinity),
    Array(1, 1, 1, 0, 0.38647623077123266, 0.0),
    Array(10, 10, 10, 10, 1.0, 1.0),
    Array(51, 43, 22, 92, 1.462626e-7, (51.0 * 92) / (22 * 43))
  )
  
  @Test(dataProvider = "chisq")
  def chisq(a: Int, b: Int, c: Int, d: Int, pValue: Double, oddsRatio: Double) {
      val r = eval(invoke("chisq", a, b, c, d)).asInstanceOf[Row]
      assert(D0_==(pValue, r.getDouble(0)))
      assert(D0_==(oddsRatio, r.getDouble(1)))
  }
```

Comment (catoverdrive) [reply src/test/scala/is/hail/expr/ir/MathFunctionsSuite.scala]: yeah. I think the consistency between tests is nice, but I don't have strong feelings outside of that.

**Comment by jbloom22** [comment]
Cool, I made the switch.

**Pull request merged**